File: oddSumSequence.cpp
int oddSumSequence(std::vector < int > v) {
    map < string, int > map;
    stringstream ss;
    copy(v.begin(), v.end(), ostream_iterator < int > (ss, "-"));
    string s = ss.str();

    map[s] = 0;
    int cnt = 0;
    vector < int > temp;

    while (true) {

        cnt++;
        int sum = 0;

        for (int i = 0; i < v.size(); i++) {
            if (v.at(i) % 2 != 0) sum += v.at(i);
            temp.push_back(sum);
        }

        stringstream s2;
        copy(temp.begin(), temp.end(), ostream_iterator < int > (s2, "-"));
        string s3 = s2.str();

        if (map.count(s3) > 0) {
            return map[s3];
        }
        map[s3] = cnt;
        v.clear();
        v = temp;
        temp.clear();
    }

    return 0;
}

File: oddSumSequence.js
function oddSumSequence(arr) {
    let map = new Map();
    map.set(arr.join('-'), 0);

    let newArr = [];
    let count = 0;

    while (true) {
        count++;
        let sum = 0;
        for (let i = 0; i < arr.length; i++) {
            if (arr[i] % 2 !== 0) sum += arr[i];
            newArr.push(sum);
        }

        if (map.has(newArr.join('-'))) {
            return map.get(newArr.join('-'));
        } else {
            map.set(newArr.join('-'), count);
            arr = newArr;
            newArr = [];
        }
    }
}

oddSumSequence = a => {
    hash = {}
    hash[a.join('-')] = 0

    r = []
    c = 0

    while (true) {
        c++
        s = 0

        a.map(e => {
            s += e % 2 != 0 ? e : 0
            r.push(s)
        })

        if (hash[r.join('-')] != undefined) return hash[r.join('-')]

        hash[r.join('-')] = c
        a = r
        r = []

    }
}